pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -312,6 +313,32 @@ func (c *Cluster) reconcileServices(activeDaemon string) error {
 		}
 	}
 
+	return c.updateServiceSelectors(activeDaemon)
+}
+
+// Make a best effort to update the services that have been labeled for being updated
+// when the mgr has changed. They might be services for node ports, ingress, etc
+func (c *Cluster) updateServiceSelectors(activeDaemon string) error {
+	selector := metav1.ListOptions{LabelSelector: "app=rook-ceph-mgr"}
+	services, err := c.context.Clientset.CoreV1().Services(c.clusterInfo.Namespace).List(context.TODO(), selector)
+	if err != nil {
+		return errors.Wrap(err, "failed to query mgr services to update")
+	}
+	for i, service := range services.Items {
+		// Update the selector on the service to point to the active mgr
+		if service.Spec.Selector[controller.DaemonIDLabel] == activeDaemon {
+			logger.Infof("no need to update service %q", service.Name)
+			continue
+		}
+		// Update the service to point to the new active mgr
+		service.Spec.Selector[controller.DaemonIDLabel] = activeDaemon
+		logger.Infof("updating selector on mgr service %q to active mgr %q", service.Name, activeDaemon)
+		if _, err := c.context.Clientset.CoreV1().Services(c.clusterInfo.Namespace).Update(c.clusterInfo.Context, &services.Items[i], metav1.UpdateOptions{}); err != nil {
+			logger.Errorf("failed to update service %q. %v", service.Name, err)
```

Comment:
Is just logging part of the "best-effort"? Why not error here and retry?

Reply:
The services that are being updated here are not owned by Rook, they are created by the user outside Rook's control. Since they aren't owned by Rook it is too risky IMO to fail the reconcile if they fail to update.

Reply:
What's the side effect if the update fails? Is it worse than failing and less worse?

Reply:
If there is some unexpected type of service that can't be updated and the mgr failed the reconcile continuously, it would prevent OSDs from reconciling. The side effect of the update failing is that the service would be pointing to the standby mgr, which will not work.

Reply:
Ok, but is it worth reconciling the OSDs if the wrong mgr is used? So at end, nothing work and we have no way to recover/retry from this?